cdap-proto/src/main/java/io/cdap/cdap/proto/security/Principal.java

```diff
@@ -66,6 +72,11 @@ public String getKerberosPrincipal() {
     return kerberosPrincipal;
   }
 
+  @Nullable
+  public String getCredential() {
```

Comment:
Where is it used? Also it would be great to understand what is stored here

Reply:
The principal is passed to the authorization extension via the AuthorizerEnforcer interface in [1]. This string less a strict variable and more a flexible variable which can hold different types depending on what kind of UserIdentityExtractor was used to extract the user identity. For example, somebody may want to pass a username-password combination to the backend, whereas others may want to pass an access token.

[1]: https://github.com/cdapio/cdap/blob/51d4ab7c9e1a797f643acceb6f386c23aa004bfa/cdap-security-spi/src/main/java/io/cdap/cdap/security/spi/authorization/AuthorizationEnforcer.java#L54